Set Textdirection based on input language

osaxma opened the pull request:
- closes #124 
  - Exposes textDirection property of `SelectableText`. 
  - In `paragraphBuilder`, textDirection is set to `TextDirection.rtl` if the first non-whitespace character of a paragraph is from an RTL language. Everything else is `TextDirection.ltr`.


Result as seen in the added demo:


<img width="702" alt="Screen Shot 2021-04-14 at 5 50 05 PM" src="https://user-images.githubusercontent.com/46427323/114731407-66a28600-9d4a-11eb-879b-452cee5ef498.png">

matthew-carroll commented:
Hi @osaxma - a lot of things are moving around right now as we prepare for an initial pre-1.0 release. As a result, we won't have time to vet an approach to RTL support right now. It may be something we focus on for the next release. You can leave this PR here, but it will likely need to be re-written when the time comes to implement RTL support.

matthew-carroll commented:
@osaxma it looks like this PR is out of date. Would you like to try to update it, or would you prefer to close it and have me introduce text direction?

osaxma commented:
Hi @matthew-carroll 

We might need to close this one because it does not account for the following and I'm not sure how you'd prefer to handle them:
- Keyboard arrows (left and right) are not working correctly (they move the cursor in the opposite direction for RTL). 
- SuperTextField. 
- List items -- in RTL, the dot or number should be on the right side of the text and the entire item is aligned right. 

Please let me know if you need any additional information.

osaxma commented:
Hi @matthew-carroll 

I had sometime and I updated the code to catch up with the main branch. 

The three issues that I mentioned in my previous comment still exists (the third one can be seen in the following screenshot). 

<img width="632" alt="Screen Shot 2021-06-06 at 10 25 20 PM" src="https://user-images.githubusercontent.com/46427323/120937439-162a2200-c716-11eb-806e-6a7aa7fbc7f4.png">

btw, I'm not sure why the test is failing but it appears to be related to the demo that I added in the example project.

matthew-carroll commented:
@osaxma can you rebase against `main` to get the fixes for the website deployment, and also can you resolve the conflicting file?

osaxma commented:
@matthew-carroll done.

matthew-carroll commented:
@osaxma we added a formatting check for line length. You can see it as the failing check right now. Can you reformat with 120 length lines? I think then we should be able to merge this in.

osaxma commented:
@matthew-carroll done -- fixed formatting issues.

matthew-carroll reviewed (approved):
LGTM
